app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/EateryDetailViewModel.kt

```diff
@@ -11,107 +8,169 @@ import com.cornellappdev.android.eatery.data.models.Event
 import com.cornellappdev.android.eatery.data.repositories.EateryRepository
 import com.cornellappdev.android.eatery.data.repositories.UserPreferencesRepository
 import com.cornellappdev.android.eatery.data.repositories.UserRepository
+import com.cornellappdev.android.eatery.ui.components.general.MenuCategoryViewState
+import com.cornellappdev.android.eatery.ui.components.general.MenuItemViewState
+import com.cornellappdev.android.eatery.ui.components.general.toMenuCategory
 import com.cornellappdev.android.eatery.ui.viewmodels.state.EateryApiResponse
+import com.cornellappdev.android.eatery.util.fromOffsetToDayOfWeek
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.combine
-import kotlinx.coroutines.flow.map
-import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
+import java.time.LocalDateTime
 import javax.inject.Inject
 
+
+sealed class EateryDetailViewState {
+    object Loading : EateryDetailViewState()
+
+    data class Loaded(
+        val mealToShow: MealViewState,
+        val eatery: Eatery,
+        val isFavorite: Boolean,
+        val weekdayIndex: Int,
+    ) : EateryDetailViewState() {
+        val mealTypeIndex: Int = eatery.getTypeMeal(weekdayIndex.fromOffsetToDayOfWeek())
+            ?.indexOfFirst { it.first == mealToShow.description } ?: 0
+    }
+
+    data class Error(val message: String) : EateryDetailViewState()
+}
+
+
+data class MealViewState(
+    val startTime: LocalDateTime?,
+    val endTime: LocalDateTime?,
+    val menu: List<MenuCategoryViewState>?,
+    val description: String?,
+)
+
+fun MealViewState.toEvent(): Event =
+    Event(
+        description = description,
+        startTime = startTime,
+        endTime = endTime,
+        menu = menu?.map { it.toMenuCategory() }?.toMutableList()
+    )
```

Comment:
Lol, we can just put this as a regular `fun` inside of `{ ... }` brackets of the `MealViewState` class instead of making it an extension function right? Just makes it a bit more readable I think.

Reply:
lol I'm so used to making extension functions that I forget when I define my own class I can just add a function to it 💀 good call out here.